driver-core/build.gradle

```diff
@@ -30,18 +30,26 @@ ext {
 // Add native-image.properties
 sourceSets.main.resources.srcDirs = ['src/resources']
 
+configurations {
+    consumableTestRuntimeOnly {
+        extendsFrom testRuntimeOnly
+        canBeConsumed = true
+    }
+}
+
 dependencies {
     api project(path: ':bson', configuration: 'default')
 
     implementation "com.github.jnr:jnr-unixsocket:$jnrUnixsocketVersion", optional
     api "io.netty:netty-buffer:$nettyVersion", optional
     api "io.netty:netty-transport:$nettyVersion", optional
-    implementation "io.netty:netty-handler:$nettyVersion", optional
+    api "io.netty:netty-handler:$nettyVersion", optional
```

Comment:
This changes the dependency in the [published POM](https://search.maven.org/artifact/org.mongodb/mongodb-driver-core/4.3.0-beta3/jar) from optional runtime to optional compile time.